hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -236,7 +236,7 @@ private void execute() {
     }
 
     private boolean isNotAllowedToRetryOnSelection(Throwable exception) {
-        if (isBindToSingleConnection() && exception instanceof IOException) {
+        if (isBindToSingleConnection() && exception instanceof WrongTargetException) {
```

Comment:
Do you only want to deal with WrongTargetException or with every RetryableHazelcastException? I don't understand the flow sufficiently.